### PR TITLE
Fixes #1838 - Separates the python dependencies into prod and dev

### LINKED ISSUE #1838: Separate requirements.txt into dev-req.txt and production-req.txt
Currently we have a unique requirements.txt
Some of the modules are not useful once the application is deployed in production. Let's separate these in two files, so that when we deploy we only consider what we really need for running the application.

### PR DESCRIPTION
**Pull request by karlcow**
This PR fixes issue #1838

## Proposed PR background

This is the first step of properly separating the production and the development environment. Once this is done, we can think of ways to modify the fabfile so it deploys only the prod environment. 

r? @miketaylr

### CODE REVIEW

**Comment by karlcow** [comment]
This also requires to have #2829 merged BEFORE this one.

File: run.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
 ==============================================
 It seems like you don't have all dependencies.
 Please re-run:
-    pip install -r config/requirements.txt
+    pip install -r config/dev-requirements.txt
```
Comment (miketaylr) [inline run.py]: This should say `requirements-dev.txt`.

File: config/requirements-dev.txt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# This is for dev purpose
+-r requirements.txt
```
Comment (miketaylr) [inline config/requirements-dev.txt]: cool, I didn't know you could do this.

https://pip.pypa.io/en/stable/reference/pip_install/#requirements-file-format

File: run.py (hunk 1 above)
Comment (karlcow) [reply run.py]: 🤦‍♂️ thanks.

**Pull request merged**